# Rickettsiosis - Data Analysis

### Import libraries

In [1]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns

### Loading data

In [2]:
DATAPATH = '../data/csv/data.csv'

df = pd.read_csv(DATAPATH)    # reading data

In [3]:
df.head()

state  year  week  week_cases  m_accum  f_accum  cases_accum
0       Aguascalientes  2013     2         NaN      NaN      NaN          NaN
1      Baja California  2013     2         NaN      NaN      NaN          3.0
2  Baja California Sur  2013     2         NaN      NaN      NaN          NaN
3             Campeche  2013     2         NaN      NaN      NaN          NaN
4             Coahuila  2013     2         NaN      NaN      NaN          1.0

### Data cleaning 

In [4]:
df.shape

(7008, 7)

Se observa que el dataframe (tabla) contiene 7008 filas y 7 columnas.

In [5]:
df.columns

Index(['state', 'year', 'week', 'week_cases', 'm_accum', 'f_accum',
       'cases_accum'],
      dtype='object')

Para este proyecto se utiliza el [dataset](https://github.com/roemvaar/paper-rickettsiosis/blob/master/data/csv/data.csv). Los datos contienen los siguientes parámetros:

* state : Estado
* year: Año
* week: Número de semana
* week_cases: Casos en esa semana 
* m_accum: Accumulado masculino
* f_accum: Acumulado femenino
* cases_accum: Casos acumulados


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7008 entries, 0 to 7007
Data columns (total 7 columns):
state          7008 non-null object
year           7008 non-null int64
week           7008 non-null int64
week_cases     294 non-null float64
m_accum        1162 non-null float64
f_accum        1233 non-null float64
cases_accum    748 non-null float64
dtypes: float64(4), int64(2), object(1)
memory usage: 383.3+ KB


Existen algunas columnas con valores faltantes.

In [7]:
df.describe()    # basic descriptive analysis of numerical data

year         week  week_cases      m_accum      f_accum  \
count  7008.000000  7008.000000  294.000000  1162.000000  1233.000000   
mean   2014.648402    25.812785    2.714286     6.249570     8.622060   
std       1.245953    15.216634    4.176155    13.241207    15.940735   
min    2013.000000     1.000000    1.000000     1.000000     1.000000   
25%    2014.000000    12.000000    1.000000     1.000000     2.000000   
50%    2015.000000    25.000000    1.000000     2.000000     3.000000   
75%    2016.000000    39.000000    3.000000     5.000000     8.000000   
max    2017.000000    53.000000   33.000000   137.000000   171.000000   

       cases_accum  
count   748.000000  
mean     44.239305  
std     103.366494  
min       1.000000  
25%       2.000000  
50%       5.000000  
75%      35.000000  
max     988.000000

- Se observa que los datos vienen de los años 2013 hasta 2017.




In [8]:
df.describe(include=['object'])    # statistics on non-numerical features

state
count      7008
unique       33
top     Morelos
freq        219

**Sorting by column week_cases**

In [9]:
df.sort_values(by=['week_cases'], ascending=False).head(50)

state  year  week  week_cases  m_accum  f_accum  \
4641      Baja California  2015    44        33.0    101.0    139.0   
4609      Baja California  2015    43        33.0     85.0    122.0   
4577      Baja California  2015    42        25.0     76.0     98.0   
4673      Baja California  2015    45        22.0    110.0    152.0   
4705      Baja California  2015    46        21.0    125.0    160.0   
4481      Baja California  2015    39        21.0     54.0     63.0   
4513      Baja California  2015    40        21.0     62.0     76.0   
4449      Baja California  2015    38        16.0     48.0     48.0   
4737      Baja California  2015    47        14.0    130.0    169.0   
4545      Baja California  2015    41        11.0     67.0     82.0   
4345               Sonora  2015    34        10.0     49.0     45.0   
6361               Sonora  2016    45        10.0     14.0     12.0   
4769      Baja California  2015    48         9.0    137.0    171.0   
4193      Baja California  2015    30         9.0     22.0     18.0   
6393               Sonora  2016    46         9.0     17.0     18.0   
4161      Baja California  2015    29         8.0     17.0     14.0   
4353      Baja California  2015    35         8.0     34.0     33.0   
2489               Sonora  2014    27         8.0      8.0     25.0   
4257      Baja California  2015    32         7.0     31.0     22.0   
4417      Baja California  2015    37         7.0     40.0     41.0   
4385      Baja California  2015    36         7.0     37.0     37.0   
1697      Baja California  2014     3         7.0      3.0      6.0   
4281               Sonora  2015    32         6.0     42.0     35.0   
4441               Sonora  2015    37         6.0     52.0     40.0   
4505               Sonora  2015    39         6.0     60.0     47.0   
4121               Sonora  2015    27         6.0     23.0     23.0   
4665               Sonora  2015    44         6.0     76.0     70.0   
1729      Baja California  2014     4         6.0      5.0     10.0   
4537               Sonora  2015    40         5.0     62.0     50.0   
4313               Sonora  2015    33         5.0     44.0     38.0   
2498  Baja California Sur  2014    28         5.0      3.0     13.0   
3385               Sonora  2015     3         5.0      2.0      3.0   
4377               Sonora  2015    35         5.0     52.0     49.0   
2817      Baja California  2014    38         5.0     12.0     27.0   
4225      Baja California  2015    31         5.0     26.0     19.0   
4793               Sonora  2015    48         5.0     86.0     78.0   
4409               Sonora  2015    36         5.0     47.0     39.0   
4330           Guanajuato  2015    34         4.0      2.0      6.0   
4473               Sonora  2015    38         4.0     57.0     44.0   
1924             Coahuila  2014    10         4.0      4.0      NaN   
4569               Sonora  2015    41         4.0     65.0     51.0   
3993               Sonora  2015    23         4.0     16.0     17.0   
4129      Baja California  2015    28         4.0     13.0     10.0   
4321      Baja California  2015    34         4.0     33.0     26.0   
4025               Sonora  2015    24         4.0     19.0     18.0   
4289      Baja California  2015    33         4.0     31.0     24.0   
2649               Sonora  2014    32         4.0     13.0     26.0   
4697               Sonora  2015    45         4.0     77.0     73.0   
4097      Baja California  2015    27         4.0     11.0      8.0   
2340             Coahuila  2014    23         4.0     16.0     15.0   

      cases_accum  
4641          NaN  
4609          NaN  
4577          NaN  
4673          NaN  
4705          NaN  
4481          NaN  
4513          NaN  
4449          NaN  
4737          NaN  
4545          NaN  
4345          NaN  
6361          NaN  
4769          NaN  
4193          NaN  
6393          NaN  
4161          NaN  
4353          NaN  
2489          NaN  
4257          NaN 

De la tabla se puede observar: 

- Se muestran el top 50 de la mayor cantidad de casos en una sola semana. 
- La mayor cantidad de casos en una semana fue de 33, sucedió dos veces en semanas consecutivas (sem 43 y 44) en 2015 en el estado de Baja California. 
- En Baja California durante 2015 existió un periodo durante el cual se registraron muchos casos por semana. 
- Además de Baja California, Sonora es otro estado donde existieron muchos casos en una sola semana. 


In [10]:
df['week_cases'].mean(skipna=False)

nan

In [11]:
#df['week_cases'].mean(skipna=False)
df['week_cases'].mean()

2.7142857142857144

En México hubo un promedio de 2.71 casos por semana.  *Se ignora NaN*

¿Cuántos casos por semana en promedio hubieron en Sonora?

In [12]:
#df[df['state'] == 'Sonora']['week_cases'].mean(skipna=False)
df[df['state'] == 'Sonora']['week_cases'].mean()

2.710843373493976

El promedio de casos de Sonora también es de 2.71, básicamente igual al de todo el país. *Se ignora NaN*

**Casos por semana**

In [13]:
states = df['state'].unique()    # list of states

for state in states: 
    print((state, df[df['state'] == state]['week_cases'].mean()))

('Aguascalientes', nan)
('Baja California', 6.826923076923077)
('Baja California Sur', 1.4193548387096775)
('Campeche', 1.0)
('Coahuila', 1.4893617021276595)
('Colima', 2.0)
('Chiapas', 1.0)
('Chihuahua', 1.25)
('Distrito Federal', nan)
('Durango', 1.0)
('Guanajuato', 1.375)
('Guerrero', 1.3333333333333333)
('Hidalgo', nan)
('Jalisco', 1.0)
('México', nan)
('Michoacán', 1.3333333333333333)
('Morelos', 1.1666666666666667)
('Nayarit', nan)
('Nuevo León', 1.1666666666666667)
('Oaxaca', nan)
('Puebla', nan)
('Querétaro', nan)
('Quintana Roo', 2.0)
('San Luis Potosí', 1.0)
('Sinaloa', 1.3529411764705883)
('Sonora', 2.710843373493976)
('Tabasco', nan)
('Tamaulipas', 1.3333333333333333)
('Tlaxcala', nan)
('Veracruz', 1.0)
('Yucatán', nan)
('Zacatecas', 3.0)
('Ciudad de México', nan)


Se muestra el promedio de casos por semana en cada uno de los estados de la república.

In [14]:
df.apply(np.max)

state          Zacatecas
year                2017
week                  53
week_cases            33
m_accum              137
f_accum              171
cases_accum          988
dtype: object

Keep working based on:
- https://github.com/roemvaar/data-science-projects/blob/master/911-calls-data-capstone-project.ipynb
- https://github.com/roemvaar/data-science-projects/blob/master/finance-project.ipynb
- https://towardsdatascience.com/hitchhikers-guide-to-exploratory-data-analysis-6e8d896d3f7e?gi=455440dae33a- 
- https://www.kaggle.com/ekami66/detailed-exploratory-data-analysis-with-python

### Preguntas 

Años y frecuencia

X meses y  y frecuencia